lib/src/lib/components/form-fields/button-toggle-field/button-toggle-field.component.ts

```diff
@@ -12,45 +12,37 @@ export class ButtonToggleFieldComponent extends FormComponent<FormFieldButtonTog
   @HostBinding('class')
   public classList = 'lab900-form-field';
 
-  public value: any;
+  public currentValue: any;
 
   public constructor(translateService: TranslateService) {
     super(translateService);
     setTimeout(() => {
       if (this.group?.controls) {
-        this.setValue(this.group.controls[this.fieldAttribute].value);
+        this.currentValue = this.group.controls[this.fieldAttribute].value;
         this.addSubscription(
           this.group.controls[this.fieldAttribute].valueChanges,
-          (value: any) => setTimeout(() => this.setValue(value))
+          (value: any) => setTimeout(() => (this.currentValue = value))
         );
       }
     });
   }
 
-  private setValue(value: any): void {
-    this.value = this.options?.readonlyDisplay
-      ? this.options?.readonlyDisplay(this.group.value)
-      : value;
-  }
-
   public get label(): string {
     const option = this.options.buttonOptions.find(
-      (o) => o.value === this.value
+      (o) => o.value === this.currentValue
     );
-    return option?.label ?? `${this.value}`;
+    return this.options?.readonlyDisplay
+      ? this.options?.readonlyDisplay(this.group.value)
+      : option?.label;
```

Comment:
Does this not create a mismatch between value and what you see on the screen?

Reply:
How so? You either show the readonlyDisplay defined in the config, or you show the label of the button with the selected value.